communication/mavlink_waypoint_handler.cpp

```diff
@@ -114,34 +106,20 @@ void Mavlink_waypoint_handler::send_waypoint(Mavlink_waypoint_handler* waypoint_
         {
             waypoint_handler->sending_waypoint_num_ = packet.seq;
             if (waypoint_handler->sending_waypoint_num_ < waypoint_handler->waypoint_count_)
             {
-                //  Prototype of the function "mavlink_msg_mission_item_send" found in mavlink_msg_mission_item.h :
-                // mavlink_msg_mission_item_send (  mavlink_channel_t chan, uint8_t target_system, uint8_t target_component, uint16_t seq,
-                //                                  uint8_t frame, uint16_t command, uint8_t current, uint8_t autocontinue, float param1,
-                //                                  float param2, float param3, float param4, float x, float y, float z)
-                mavlink_message_t _msg;
-                mavlink_msg_mission_item_pack(sysid,
-                                              waypoint_handler->mavlink_stream_.compid(),
-                                              &_msg,
-                                              msg->sysid,
-                                              msg->compid,
-                                              packet.seq,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].frame,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].command,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].current,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].autocontinue,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param1,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param2,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param3,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].param4,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].x,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].y,
-                                              waypoint_handler->waypoint_list[waypoint_handler->sending_waypoint_num_].z);
-                waypoint_handler->mavlink_stream_.send(&_msg);
+                uint8_t isCurrent = 0;
+                if (    waypoint_handler->sending_waypoint_num_ == waypoint_handler->current_waypoint_index_ && // This is the current waypoint
+                        !waypoint_handler->navigation_.waiting_at_waypoint())                                   // And we are en route
```

Comment:
As I understood the previous code, the current variable can have 2 states: 0 or 1. There can only be one waypoint that is current, but if the drone is not currently travelling towards a waypoint, all waypoints should have 0 as current.

I could be wrong about this as this was one of the parts of the code that had redundant variables that were updated in many different locations.